pkg/internal/parser.go

```diff
@@ -0,0 +1,174 @@
+type interfaceWrapper struct {
+	// name of the interface provided from the input flag.
+	InterfaceName string
+	// input path where types are defined.
+	InputPath string
+	// APIs wrap each of the type
+	APIs []api
+	// writer wherein outputs are written
+	wr *io.Writer
+}
+
+// api contains info about each type
+type api struct {
+	Name    string
+	Version string
+	PkgName string
+	wr      io.Writer
+
+	PkgNameUpperFirst string
+	VersionUpperFirst string
+	NameLowerFirst    string
+}
+
+// packages stores the info used to scaffold wrapped interfaces content
+type packages struct {
+	Name              string
+	APIPath           string
+	ClientPath        string
+	NameUpperFirst    string
+	VersionUpperFirst string
+	Version           string
+	wr                io.Writer
+}
+
+// NewInterfaceWrapper returns a interfaceWrapper which can fill the templates to wrtie clientset wrappers.
+func NewInterfaceWrapper(interfaceName string, inputPath string, gvs []gentype.GroupVersions, wr io.Writer) (*interfaceWrapper, error) {
+	apis := groupVersionsToApis(gvs)
+	return &interfaceWrapper{
+		InputPath:     inputPath,
+		InterfaceName: interfaceName,
+		APIs:          apis,
+		wr:            &wr,
+	}, nil
+}
+
+// TODO: this could be converted to an interface, wherein each sub-generator has a writeContent method.
+func (w *interfaceWrapper) WriteContent() error {
+	templ, err := template.New("wrapper").Parse(wrappedInterfacesTempl)
+	if err != nil {
+		return err
+	}
+	return templ.Execute(*w.wr, w)
+}
+
+// groupVersionToApis converts a list of types.GroupVersions to api type which can then be used for
+// templating.
+// Note: `Versions` in type.GroupVersions is assumed to contain only one version for now.
+func groupVersionsToApis(gvs []gentype.GroupVersions) []api {
+	result := make([]api, 0)
+
+	for _, gv := range gvs {
+		// this shouldn't happen, we would error out in this condition while validating flags.
+		if len(gv.Versions) <= 0 {
+			continue
+		}
+		a := &api{
+			Name:    gv.Group.String(),
+			Version: string(gv.Versions[0].Version),
+			PkgName: gv.PackageName,
+		}
+		a.setCased()
+		result = append(result, *a)
+	}
+	return result
+}
+
+func (a *api) setCased() {
+	a.PkgNameUpperFirst = upperFirst(a.PkgName)
+	a.VersionUpperFirst = upperFirst(a.Version)
+	a.NameLowerFirst = lowerFirst(a.Name)
+}
+
+func (p *packages) setCased() {
+	p.NameUpperFirst = upperFirst(p.Name)
+	p.VersionUpperFirst = upperFirst(p.Version)
+}
+
+// lowerFirst sets the first alphabet to lowerCase.
+func lowerFirst(s string) string {
+	return strings.ToLower(string(s[0])) + s[1:]
+}
+
+// upperFirst sets the first alphabet to upperCase/
+func upperFirst(s string) string {
+	return strings.ToUpper(string(s[0])) + s[1:]
+}
+
+// NewPacakges returns a new packages instance which is used to write wrapper content.
+func NewPacakges(root *loader.Package, apiPath, clientPath, version, group string, wr io.Writer) (*packages, error) {
```

Comment:
```suggestion
func NewPackages(root *loader.Package, apiPath, clientPath, version, group string, wr io.Writer) (*packages, error) {
```

Reply:
Doesn't look there are any error flows - might change the signature to just return `*packages`